design/20220722-publicbundle.md

```diff
@@ -0,0 +1,279 @@
+var certFiles = []string{
+	"/etc/ssl/certs/ca-certificates.crt",                // Debian/Ubuntu/Gentoo etc.
+	"/etc/pki/tls/certs/ca-bundle.crt",                  // Fedora/RHEL 6
+	"/etc/ssl/ca-bundle.pem",                            // OpenSUSE
+	"/etc/pki/tls/cacert.pem",                           // OpenELEC
+	"/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", // CentOS/RHEL 7
+	"/etc/ssl/cert.pem",                                 // Alpine Linux
+}
+```
+
+It's a chore to manually specify a location for a trust store, and it's unfamiliar and error-prone
+for many users. However, if a user instead overwrites any of these files with their own
+trust-manager-generated certificate trust bundle they'll currently overwrite their public trust bundle
+provided by their OS / container base image.
+
+Unless the user's application only talks to services which the user also controls, what users are
+likely to really want in most cases is a bundle containing both the publicly trusted certs _and_
+the user's own roots.
+
+It's possible in most distros to append to the cert bundle, but that usually involves running some
+command to generate the final bundle, which would change how the user invokes their application
+inside their container.
+
+Ideally, the whole bundle a user requires would simply be available at one of the above locations.
+
+In Kubernetes, that today likely means overwriting the system trust bundle with an entire bundle
+at the point when the container is created. That implies that the Target which trust-manager creates
+must be completed before the container starts, and that in turn implies a need for the public
+trust bundles users rely on to be available as sources for trust-manager.
+
+### Goals
+
+- A user configuring a `Bundle` can add a public trust bundle at least as easily as they can add certs from a `ConfigMap` or `Secret`
+- It's clear to users exactly what version of the public bundle is used at any given time, for auditing purposes
+- A user whose application must use both public and private CAs can confidently overwrite their "system" trust store in a container using a trust `Bundle`
+- A user can update any bundle in use with any version of the trust-manager controller and be confident that it will work
+
+### Non-Goals
+
+- Editing/pruning the public trust bundle (out of scope for the initial feature but might be desirable later)
+- Supporting [OS-specific](https://superuser.com/questions/437330/how-do-you-add-a-certificate-authority-ca-to-ubuntu) certificate update mechanisms
+- Supporting a plethora of options for sources of public trust bundles; we might want this in the future but we should start with one
+- Dynamically updating the public bundle (users will have to explicitly choose to upgrade, at least for now)
+
+## Proposal
+
+We propose that the cert-manager project becomes a distributor of its own public trust bundle, which at
+least to start will be a carbon copy of the Debian `ca-certificates` package.
```

Comment:
It would be great to have blocklisting over this - the issue with the distro trust bundles is they include too much, including the ones that are clearly iffy. I'd like to be able to say "debian, minus foo and bar"

Reply:
Thanks for the comment, really appreciate it!

I fully agree that this would be great functionality for us to add down the road. I'm personally super invested in making trust-manager handle these situations!